modules/libcom/src/misc/epicsExport.h

```diff
@@ -75,12 +71,24 @@ typedef void (*REGISTRAR)(void);
  *
  * \param typ Object's data type.
  * \param obj Object's name.
- *
- * \note C++ code needs to wrap with @code extern "C" { } @endcode
  */
+#ifdef __cplusplus
+namespace epics {
+    template<typename T>
+    static inline T* ignore_cv_qualifier(volatile T* x) { return const_cast<T*>(x); }
+    template<typename T>
+    static inline T* ignore_cv_qualifier(const T* x) { return const_cast<T*>(x); }
+    template<typename T>
+    static inline T* ignore_cv_qualifier(T* x) { return x; }
```

Comment:
A note for the future.  `ignore_cv_qualifier()` has the same function as the c++11 [`std::remove_cv`](https://en.cppreference.com/w/cpp/types/remove_cv), although a somewhat different invocation.  Since this would (I think) depend on `decltype()`, it can't yet be used directly.

For the present, I think `ignore_cv_qualifier()` may actually not be inline'd in all cases (compiler versions) since `inline` is only a hint.  The c++11 `constexpr` modifier would ensure this, but we're not c++11 yet...

Maybe just keeping the heavy handed `(typ *) (char *)` casting would be a more straightforward solution/workaround until then?

Reply:
> `ignore_cv_qualifier()` has the same function as the c++11 `std::remove_cv`, although a somewhat different invocation. 

No.  `std::remove_cv` can only be uses to compare types qualifier-less. But not to cast pointers.

Reply:
> Maybe just keeping the heavy handed (typ *) (char *) casting would be a more straightforward solution/workaround until then?

According to my tests, that seems to be the only way to make the CI tests succeed. I still do not understand, why only half of the test architectures in don't work and why every test always succeeds on my local system.